snapwell/snapecl.py

```diff
@@ -130,54 +104,81 @@ def interpolate(grid, k_above, col, thresh):
+def first_swat_below_treshold(column, threshold=0.7):
+    """
+    finds the first (i,j,k,a,swat) tuple with swat less than treshold.
+    If there is no such tuple, returns None.
+    :param column: List of (x,y,z,a,swat) tuples.
+    """
+    for idx, (_, _, _, _, swat) in enumerate(column):
+        if swat < (threshold + 0.0001):
+            return idx
+    return None
 
-    This function returns a pair (owc, tvd) where owc is the approximate
-    (linear interpolated) OWC for (x,y) and tvd is the lowest cell center
-    above owc_exact - owc_offset.
 
+def interpolate_owc(grid, col, k_above, threshold=0.7):
     """
+    This function returns  the approximate
+    (linear interpolated) OWC for (x,y).
 
-    col, _ = activeCellColumn(grid, owc_kw, x, y, z)
-    if not col:
-        raise ValueError(
-            "Could not find cell column for (%.2f, %.2f, %.2f) in grid." % (x, y, z)
-        )
+    :param col: List of (i,j,k,a,s) tuples to interpolate over.
+    :param grid: The grid the column belongs to.
+    :param k_above:index of first cell whose center is above owc.
+    """
 
-    current_swat = -1
-    # Called inv_k because it is inv_k = nz - k
-    for inv_k, (_, _, _, active, sat) in enumerate(col):
-        if active >= 0:
-            current_swat = sat
-            break
-    # inv_k is first active index from below.
-    if current_swat < 0:
-        raise ValueError("No active cell in column for (%.2f,%.2f,%.2f)" % (x, y, z))
-
-    while current_swat >= (thresh + 0.0001):
-        inv_k += 1
-        if inv_k >= len(col):
-            raise ValueError("No OWC in column for (%.2f,%.2f,%.2f)" % (x, y, z))
-        current_swat = col[inv_k][4]
-
-    a_idx = col[inv_k][3]  # col[idx] contains x,y,z,a,s
-    cell_center = grid.get_xyz(active_index=a_idx)[2]  # the z value of cell center
-
-    if inv_k > 0:
-        owc_exact = interpolate(grid, inv_k, col, thresh)
-    else:
-        owc_exact = grid.get_xyz(active_index=a_idx)[2]
+    a_idx = col[k_above][3]  # col[idx] contains x,y,z,a,s
+
+    if k_above > 0:
+        return interpolate(grid, k_above, col, threshold)
+
+    return grid.get_xyz(active_index=a_idx)[2]
+
+
+def find_center_z(grid, column, height):
+    """
+    Given list of (i,j,k,a,swat) tuples giving indecies of active cells in
+    the grid. Returns the center point z value of a cell in the grid above the
+    given height.  Returns None if cell center above the last column center
+    """
+    for idx, (_, _, _, a_idx, _) in enumerate(column):
+        cell_center = grid.get_xyz(active_index=a_idx)[2]  # the z value of cell center
+        if height >= cell_center:
+            return cell_center
+    return None
 
+
+def find_owc(grid, owc_kw, x, y, z, threshold=0.7, owc_offset=0.5):
+    """Given a grid, owc_kw, x, y, z, find the OWC Z s.t. owc_kw(x,y,Z)=thresh."""
+    col = active_cell_column(grid, owc_kw, x, y, z)
+    if not col:
+        logging.warning(
```

Comment:
Use new style formatting?

Reply:
logging doesn't like it as it wants values to be lazily inserted in order to do formatting correctly.